agent/engine/docker_task_engine.go

```diff
@@ -955,13 +955,7 @@ func (engine *DockerTaskEngine) startContainer(task *apitask.Task, container *ap
 		}
 	}
 	startContainerBegin := time.Now()
-
-	ctxTimeoutStartContainer := container.GetStartTimeout()
-	if ctxTimeoutStartContainer <= 0 {
-		ctxTimeoutStartContainer = engine.cfg.ContainerStartTimeout
-	}
-
-	dockerContainerMD := client.StartContainer(engine.ctx, dockerContainer.DockerID, ctxTimeoutStartContainer)
+	dockerContainerMD := client.StartContainer(engine.ctx, dockerContainer.DockerID, engine.cfg.ContainerStartTimeout)
```

Comment:
is my understanding correct?
container.GetStartTimeout() (returns StartTimeout): for container dependency timing out container start
engine.cfg.ContainerStartTimeout: configured value for docker API timeout; this was used as default for above prior to this fix - if so, what about it was the problem?

Reply:
Yes, initially we thought that we will utilize this timeout as timeout for Docker API startContainer timeout as well to check whether the dependency container has timed out or not for container ordering. But now that we think about it, it might become a problem. For example,

If container A depends on container B and the condition is 'SUCCESS', the container A cannot start before B exits successfully. If B takes 2 days to exit, then StartTimeout will roughly be 2 days. Now suppose, Docker API StartContainer call fails, the customer will get to know about it after 2 days. That is why, this change is made in order to fail fast if startContainer Docker API call fails.

Related PR: https://github.com/aws/amazon-ecs-agent/pull/1878